<launch>
    <arg name="planning_scene_config" default="/root/ocla/catkin_ws/src/grinding_motion_routines/config/UR_grinding_demo/planning_scene_config.yaml"/>
    <arg name="motion_config" default="/root/ocla/catkin_ws/src/grinding_motion_routines/config/UR_grinding_demo/motion_config.yaml"/>
    <arg name="rosbag_record" default="false"/>
    <arg name="rosbag_name" default=""/>

    <node name="vision_grinding" pkg="grinding_motion_routines" type="ICRA2023_vision_grinding.py" output="screen">    
        <param name="move_group_name" value="manipulator" />
        <param name="grinding_eef_link" value="pestle_tip" />
        <param name="gathering_eef_link" value="spatula_tip" />
        <rosparam command="load" file="$(arg planning_scene_config)"/>
        <rosparam command="load" file="$(arg motion_config)"/>
    </node>


    <group if="$(arg rosbag_record)">
        <node pkg="rosbag" type="record" name="rosbag_record"
        args='record -a -o $(arg rosbag_name) -x "/camera/depth/(.*)"'/>
    </group>
</launch>